lighthouse-core/test/gather/driver-test.js

```diff
@@ -49,6 +49,88 @@ function createMockSendCommandFn() {
   return mockFn;
 }
 
+/**
+ * Creates a jest mock function whose implementation invokes `.on`/`.once` listeners after a setTimeout tick.
+ * Closely mirrors `createMockSendCommandFn`.
+ *
+ * It is decorated with two methods:
+ *    - `mockEvent` which pushes protocol event payload for consumption
+ *    - `findListener` which asserts that `on` was invoked with the given event name and
+ *      returns the listener .
+ */
+function createMockOnceFn() {
+  const mockEvents = [];
+  const mockFn = jest.fn().mockImplementation((event, listener) => {
+    const indexOfResponse = mockEvents.findIndex(entry => entry.event === event);
+    if (indexOfResponse === -1) return;
+    const {response} = mockEvents[indexOfResponse];
+    mockEvents.splice(indexOfResponse, 1);
+    // Wait a tick because real events never fire immediately
+    setTimeout(() => listener(response), 0);
+  });
+
+  mockFn.mockEvent = (event, response) => {
+    mockEvents.push({event, response});
+    return mockFn;
+  };
+
+  mockFn.findListener = event => {
+    expect(mockFn).toHaveBeenCalledWith(event, expect.anything());
+    return mockFn.mock.calls.find(call => call[0] === event)[1];
+  };
+
+  mockFn.findListeners = event => {
+    return mockFn.mock.calls.filter(call => call[0] === event).map(call => call[1]);
+  };
+
+  return mockFn;
+}
+
+/**
+ * Transparently augments the promise with inspectable functions to query its state.
+ *
+ * @template T
+ * @param {Promise<T>} promise
+ * @return {Promise<T> & {isDone: () => boolean, isResolved: () => boolean, isRejected: () => boolean}}
+ */
+function createInspectablePromise(promise) {
```

Comment:
maybe we need another name for this? It's less creating and more making a promise inspectable

Reply:
sounds good 👍 

create, make, it's all the same to me 🤷‍♂️